syntax="proto3";

package pr_svg;

message Rgb
{
	uint32 red = 1;  
    uint32 green = 2;  
    uint32 blue = 3;
}
message Rgba
{
    uint32 red = 1;  
    uint32 green = 2;  
    uint32 blue = 3;
    double opacity = 4;
}
message Point
{
    double x = 1;  
    double y = 2; 
}
message Color
{
    oneof color_variant
    {
        bool monostate = 1;
        string color_string = 2;
        Rgb color_rgb = 3;
        Rgba color_rgba = 4;
    }
}